### ISSUE TITLE
Expose stylelint CLI opts

### ISSUE THREAD

**Issue by Undistraction** [shinnn/vscode-stylelint#209]
I'm using this lib to lint styles defined in [styled components](https://github.com/styled-components/styled-components).

After discussing this with @chinesedfan in this [issue](https://github.com/styled-components/stylelint-processor-styled-components/issues/195#issuecomment-393577155) at the the stylelint-processor-styled-components repo, it appears that the only way to have that processor lint correctly is to toggle stylelint's syntax to 'scss'. This [cannot be done](https://github.com/stylelint/stylelint/issues/2973) in a `.stylelintrc` and can only be achieved by supplying a flag to stylelint directly:

```bash
stylelint './src/**/*.js' --syntax scss
```

Currently there is no way to supply this flag when using stylelint via `vscode-stylelint`.

Please expose CLI options via a setting.

**Comment by shinnn** [comment]
Thanks for the suggestion, but I still think `syntax` option should be configurable via `.stylelintrc.*` files. Modifying the whole behavior of stylelint via an editor setting looks like a bad idea for me.

Ask stylelint maintainers to reopen https://github.com/stylelint/stylelint/issues/2973. One of the maintainers [said](https://github.com/stylelint/stylelint/issues/2973#issuecomment-350076915):

> Closing as the specific use case was addressed in https://github.com/stylelint/stylelint/pull/2975

but I don't think so.
